### PR TITLE
Creating infraction restore command

### PR DESCRIPTION
**Pull request by QuesosNachos**
Creating an infraction restore command to address #517. Forewarning, I don't have any knowledge on how infractions are marked or flagged as deleted or rescinded in the database. This PR might be completely useless, but here the infraction restoration command (if it actually works?). 

Please review and we can make changes as necessary to get this working

### CODE REVIEW

File: Modix.Data/Repositories/InfractionRepository.cs [hunk 1]
```diff
@@ -104,17 +104,27 @@ public interface IInfractionRepository
...
+        /// <returns>A <see cref="Task"/> which will complete when the operation is complete,
         /// containing a flag indicating whether the update was successful (I.E. whether the specified infraction could be found).
         /// </returns>
         Task<bool> TryRescindAsync(long infractionId, ulong rescindedById);
 
+        /// <summary>
+        /// Marks a deleted infraction as active, based on its ID
+        /// </summary>
+        /// <param name="infractionId">The <see cref="InfractionEntity.Id"/> value of the infraction to be restored.</param>
+        /// <param name="restoredById">The <see cref="UserEntity.Id"/> value of the user that is rescinding the infraction.</param>
```
Comment (Inzanit) [inline Modix.Data/Repositories/InfractionRepository.cs]: Not a major deal, but this does say "rescinding", not that the XML documentation is all too important.

File: Modix.Data/Repositories/InfractionRepository.cs [hunk 2]
```diff
@@ -267,6 +277,33 @@ public async Task<bool> TryRescindAsync(long infractionId, ulong rescindedById)
...
+            {
+                GuildId = entity.GuildId,
+                Type = ModerationActionType.InfractionRestored,
+                Created = DateTimeOffset.Now,
+                CreatedById = restoredById,
+                InfractionId = entity.Id
+            };
+
+            await ModixContext.SaveChangesAsync();
+            await RaiseModerationActionCreatedAsync(entity.RescindAction);
```
Comment (Inzanit) [inline Modix.Data/Repositories/InfractionRepository.cs]: I think we need to modify the mod-log entry here, needs to defined in https://github.com/discord-csharp/MODiX/blob/9055b02859a2a8fceff6db0dddd2d134c2ffc408/Modix.Bot/Behaviors/ModerationLoggingBehavior.cs#L107-L126

**Comment by QuesosNachos** [comment]
Yep, I have to run the create and test the migrations real quick

File: Modix.Data/Repositories/InfractionRepository.cs (hunk 1 above)
Comment (QuesosNachos) [reply Modix.Data/Repositories/InfractionRepository.cs]: Fixed

File: Modix.Data/Repositories/InfractionRepository.cs (hunk 2 above)
Comment (QuesosNachos) [reply Modix.Data/Repositories/InfractionRepository.cs]: Added

**Comment by QuesosNachos** [comment]
I've made the changes suggested in your review. Do not merge this yet as there is a still a bug with the command. The infraction search command is not seeing infractions that have been restored.

File: Modix.Data/Repositories/InfractionRepository.cs (hunk 2 above)
Comment (Exadon) [inline Modix.Data/Repositories/InfractionRepository.cs]: `await RaiseModerationActionCreatedAsync(entity.RescindAction);` appears incorrect. consider replacing it with `await RaiseModerationActionCreatedAsync(entity.RestoreAction);`

File: Modix.Data/Repositories/InfractionRepository.cs [hunk 3]
```diff
@@ -267,6 +277,33 @@ public async Task<bool> TryRescindAsync(long infractionId, ulong rescindedById)
...
         }
 
+        /// <inheritdoc />
+        public async Task<bool> TryRestoreAsync(long infractionId, ulong restoredById)
+        {
+            var entity = await ModixContext.Infractions
+                .Where(x => x.Id == infractionId)
+                .FirstOrDefaultAsync();
+
+            if ((entity == null) || (entity.RescindActionId != null))
```
Comment (Exadon) [inline Modix.Data/Repositories/InfractionRepository.cs]: `if ((entity == null) || (entity.RescindActionId != null))` appears incorrect. Consider replacing it with `if ((entity == null) || (entity.RestoreActionId != null))`

Comment (QuesosNachos) [reply Modix.Data/Repositories/InfractionRepository.cs]: Fixed

File: Modix.Data/Repositories/InfractionRepository.cs (hunk 2 above)
Comment (QuesosNachos) [reply Modix.Data/Repositories/InfractionRepository.cs]: Fixed

File: Modix.Data/Repositories/InfractionRepository.cs (hunk 3 above)
Comment (Inzanit) [reply Modix.Data/Repositories/InfractionRepository.cs]: Did you push that, @QuesosNachos I see `RescindAction` still? Unless this end hasn't updated.

**Pull request closed without merging**